18F/identity-idp
# Add "Attention with Barcode" support to mock doc auth client

Extracted from #6483

**Why**:

- To be able to test the behavior locally
- To have consistent implementation expectations across all doc auth clients

**Implementation notes:**

This does a couple things:

- Refactors the mock response class to extend `DocAuth::Response` to align with Acuant and TrueID
- Adds `attention_with_barcode?` as an abstract method to `DocAuth::Response` to ensure presence for future work of LG-5918 (#6483)

**Testing instructions:**

You should be able to pass document capture locally using this [test file](https://developers.login.gov/testing/#testing-identity-proofing):

```yaml
document:
  type: license
  first_name: Susan
  last_name: Smith
  middle_name: Q
  address1: 1 Microsoft Way
  address2: Apt 3
  city: Bayside
  state: NY
  zipcode: "11364"
  dob: 10/06/1938
  phone: +1 314-555-1212
failed_alerts:
  - name: 2D Barcode Read
    result: Attention
```
